src/optimize.ml

```diff
@@ -0,0 +1,89 @@
+open Types
+open Types.Simplified
+
+let rec optimize_expr (exp : expr) : expr =
+  match exp with
+  | I32_const x :: I32_const y :: I_binop (S32, Add) :: tl ->
+      optimize_expr (I32_const (Int32.add x y) :: tl)
+  | I32_const x :: I32_const y :: I_binop (S32, Sub) :: tl ->
+      optimize_expr (I32_const (Int32.sub x y) :: tl)
+  | I32_const x :: I32_const y :: I_binop (S32, Mul) :: tl ->
+      optimize_expr (I32_const (Int32.mul x y) :: tl)
+  | I32_const x :: I32_const y :: I_binop (S32, Div U) :: tl ->
+      optimize_expr (I32_const (Int32.div x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Add) :: tl ->
+      optimize_expr (I64_const (Int64.add x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Sub) :: tl ->
+      optimize_expr (I64_const (Int64.sub x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Mul) :: tl ->
+      optimize_expr (I64_const (Int64.mul x y) :: tl)
+  | I64_const x :: I64_const y :: I_binop (S64, Div U) :: tl ->
```

Comment:
I believe it could be possible to do something like:

```ocaml
| ((I32_const _ | I64_const _) as x) :: ((I32_const _ | I64_const _) as y) :: I_binop (nn, op) :: tl ->
  let result = Interpret.exec_ibinop [] nn op in
  begin
    match result with
    | [I32_const _ | I64_const _ as result ] -> result :: optimize_expr tl
    | _ -> assert false
  end
```

Could you try if this works ? This would make the code much shorter as you'll have a single match case for all theses binary operators cases :)